Make models

Rossi-Luciano opened the pull request:
### O que esse PR faz?
Cria um modelo para armazenamento de nomes padrões de financiadores de pesquisas.

### Onde a revisão poderia começar?
Por _commit_

### Como este poderia ser testado manualmente?
NA.

### Algum cenário de contexto que queira dar?
NA.

### Screenshots
![image](https://user-images.githubusercontent.com/41302084/177646184-138c59db-7234-4c09-b1e0-266a88cf73a1.png)
![image](https://user-images.githubusercontent.com/41302084/177646287-415cf1d1-4ac9-43d8-be0d-8aacbe160879.png)
![image](https://user-images.githubusercontent.com/41302084/177657924-bd843c40-d0db-4395-a95e-13834bb764a9.png)
![image](https://user-images.githubusercontent.com/41302084/177657977-5de82373-f4ec-4b4d-b4db-10799a9d1211.png)
![image](https://user-images.githubusercontent.com/41302084/177660579-70532bbf-b3ba-4f61-a053-bd0bf36f78be.png)
![image](https://user-images.githubusercontent.com/41302084/177660613-1cdf537f-5cec-460a-a466-17c56dc061e7.png)


### Quais são tickets relevantes?
NA.

### Referências
NA.

Rossi-Luciano committed:
Adiciona chkcsv.py para verificação de conteúdo de um arquivo CSV

Rossi-Luciano committed:
Adiciona models.py

Rossi-Luciano committed:
Adiciona __init__.py

Rossi-Luciano committed:
Adiciona admin.py

Rossi-Luciano committed:
Adiciona apps.py

Rossi-Luciano committed:
Adiciona button_helper.py

Rossi-Luciano committed:
Adiciona chkcsvfmt.fmt

Rossi-Luciano committed:
Adiciona forms.py

Rossi-Luciano committed:
Adiciona models.py

Rossi-Luciano committed:
Adiciona tests.py

Rossi-Luciano committed:
Adiciona urls.py

Rossi-Luciano committed:
Adiciona views.py

Rossi-Luciano committed:
Adiciona wagtail_hooks.py

Rossi-Luciano committed:
Adiciona __init__.py em migrations

Rossi-Luciano committed:
Adiciona 0001_initial.py em migrations

Rossi-Luciano committed:
Adiciona __init__.py ao aticle_sponsors

Rossi-Luciano committed:
Adiciona admin.py ao article_sponsors

Rossi-Luciano committed:
Adiciona apps.py ao article_sponsors

Rossi-Luciano committed:
Adiciona button_helper.py ao article_sponsors

Rossi-Luciano committed:
Adiciona chkcsvfmt.fmt ao article_sponsors

Rossi-Luciano committed:
Adiciona forms.py ao article_sponsors

Rossi-Luciano committed:
Adiciona 0001_initial.py ao article_sponsors/migrations

Rossi-Luciano committed:
Adiciona __init__.py ao article_sponsors/migrations

Rossi-Luciano committed:
Adiciona models.py ao article_sponsors

Rossi-Luciano committed:
Adiciona tests.py ao article_sponsors

Rossi-Luciano committed:
Adiciona urls.py ao article_sponsors

Rossi-Luciano committed:
Adiciona views.py ao article_sponsors

Rossi-Luciano committed:
Adiciona wagtail_hooks.py ao article_sponsors

Rossi-Luciano committed:
Adiciona __init__.py ao identified_sponsors

Rossi-Luciano committed:
Adiciona admin.py ao identified_sponsors

Rossi-Luciano committed:
Adiciona apps.py ao identified_sponsors

Rossi-Luciano committed:
Adiciona button_helper.py ao identified_sponsors

Rossi-Luciano committed:
Adiciona chkcsvfmt.fmt ao identified_sponsors

Rossi-Luciano committed:
Adiciona forms.py ao identified_sponsors

Rossi-Luciano committed:
Adiciona 0001_initial.py ao identified_sponsors/migrations

Rossi-Luciano committed:
Adiciona __init__.py ao identified_sponsors/migrations

Rossi-Luciano committed:
Adiciona models.py ao identified_sponsors

Rossi-Luciano committed:
Adiciona tests.py ao identified_sponsors

Rossi-Luciano committed:
Adiciona urls.py ao identified_sponsors

Rossi-Luciano committed:
Adiciona views.py ao identified_sponsors

Rossi-Luciano committed:
Adiciona wagtail_hooks.py ao identified_sponsors

robertatakenaka commented on standard_sponsors/chkcsvfmt.fmt:
```diff
@@ -0,0 +1,11 @@
+[ID]
+data_required=True
+type=Integer
```
@Rossi-Luciano acho que o sistema deve atribuir o ID e não o usuário que ingressa os dados com a planilha.
Então, esta coluna não seria obrigatória, na verdade, deveria ser ignorada.
E internamente, o conteúdo das linhas deveria ser verificado se está ou não registrado no sistema. Se sim, não ingressar, se não, criar novo registro. E o ID acaba sendo gerado ao criar o registro.

robertatakenaka commented on standard_sponsors/chkcsvfmt.fmt:
```diff
@@ -0,0 +1,11 @@
+type=Integer
+
+[Standard_Name]
+data_required=False
```
@Rossi-Luciano já aqui deveria ser True

robertatakenaka commented on standard_sponsors/chkcsvfmt.fmt:
```diff
@@ -0,0 +1,11 @@
+type=String
+
+[Standard_Acronym]
+data_required=False
```
@Rossi-Luciano já aqui deveria ser True

robertatakenaka commented on standard_sponsors/models.py:
```diff
@@ -0,0 +1,49 @@
+from .forms import StandardSponsorsForm, StandardSponsorsFileForm
+
+
+class SponsorNamesStandardized(CommonControlField):
```
@Rossi-Luciano incluir comentário. Não ficou explícito se é uma tabela com títulos "padronizados" ou que "foram padronizados".  Se é o primeiro caso name e acron jamais deveriam ser `null=True`. Só haveria sentido se o par fosse ingressado.

robertatakenaka commented on standard_sponsors/views.py:
```diff
@@ -0,0 +1,84 @@
+                isd.sponsor_name_std = row['Standard_Name']
+                isd.sponsor_acron_std = row['Standard_Acronym']
+                isd.creator = request.user
+                isd.save()
```
@Rossi-Luciano criar um bloco de try / exception dentro de `for`, pois se houver exceções em algum item a função inteira é interrompida

robertatakenaka commented on article_sponsors/chkcsvfmt.fmt:
```diff
@@ -0,0 +1,27 @@
+
+[Score_SEM]
+data_required=False
+type=Float
```
@Rossi-Luciano eu acho que na planilha não entrarão estes campos

robertatakenaka replied:
@Rossi-Luciano pensando melhor acho que pode haver os dois modos: um para popular a base de dados diretamente, outro para inserir os dados a serem completados / padronizados pelo sistema

robertatakenaka commented on article_sponsors/models.py:
```diff
@@ -0,0 +1,58 @@
+
+    pid = models.CharField("PID", max_length=255, null=True, blank=True)
+    sponsor_name = models.CharField("Sponsor Name", max_length=255, null=True, blank=True)
+    project_id = models.CharField("Project ID", max_length=255, null=True, blank=True)
```
@Rossi-Luciano sponsor_name deve ser obrigatório (null=False, blank=False)

robertatakenaka commented on article_sponsors/views.py:
```diff
@@ -0,0 +1,88 @@
+                isd.std_id_sem = row['Std_ID_SEM']
+                isd.score_sem = row['Score_SEM']
+                isd.creator = request.user
+                isd.save()
```
@Rossi-Luciano inserir um bloco de try / exception

Rossi-Luciano committed:
Renomeia app standard_sponsors

Rossi-Luciano committed:
Adiciona arquivos em official_sponsors

Rossi-Luciano committed:
Adiciona padrão de checagem para upload de CSV

Rossi-Luciano committed:
Adiciona button_helper

Rossi-Luciano committed:
Adiciona forms.py

Rossi-Luciano committed:
Adiciona models.py

O modelo official_sponsors ficará com somente dois parâmetros: nome e
acrônimo do financiador

Rossi-Luciano committed:
Adiciona views.py

Rossi-Luciano committed:
Adiciona wagtail_hooks.py

Rossi-Luciano committed:
Corrige padrão de checagem para article_sponsor

Rossi-Luciano committed:
Corrige os parâmetros do modelo article_sponsors

Rossi-Luciano committed:
Adiciona tratamento de exceção

Rossi-Luciano committed:
Corrige wagtail_hooks

Rossi-Luciano committed:
Corrige padrões de checagem para identified_sponsors

Rossi-Luciano committed:
Corrige identified_sponsors

Rossi-Luciano committed:
Adiciona tratamento de exceção

Rossi-Luciano committed:
Corrige wagtail_hooks

Rossi-Luciano committed:
Adiciona choices para identified_sponsors

Rossi-Luciano committed:
Adiciona script para carga de dados

Rossi-Luciano committed:
Adiciona stmodels

Rossi-Luciano committed:
Remove stmodels

Rossi-Luciano committed:
Adiciona arquivo CSV com exemplos para teste

Rossi-Luciano committed:
Adiciona campos de controle à interface

Rossi-Luciano committed:
Adiciona Git

Rossi-Luciano committed:
Adiciona apps

Rossi-Luciano committed:
Indisponibiliza campos de controle

Os campos de controle 'creator' e 'updated_by' estão retornando 'None'.

Rossi-Luciano committed:
Corrige o chkcsvfmt.fmt

Rossi-Luciano committed:
Adiciona exemplos para teste

Rossi-Luciano committed:
Modifica a chamada considerando o arquivo de teste

Rossi-Luciano committed:
Modifica load_sponsor

O objetivo é que registros cujo 'pid' já estejam no banco não sejam
duplicados, o atributo 'sponsor_name' é reprocessado caso o registro
exista.

Rossi-Luciano committed:
Adiciona a função get_official_sponsors

Essa função lê 'OfficialSponsorNames' e gera uma lista de dicionários,
utilizando a função 'make_standard_sponsor'

Rossi-Luciano committed:
Adiciona chamada para 'get_official_sponsors'

Rossi-Luciano committed:
Padroniza ordem de importação

Rossi-Luciano committed:
Adiciona os campos de controle à interface

Rossi-Luciano committed:
Modifica nomes dos containers

Rossi-Luciano committed:
Adiciona campos de controle à interface

Rossi-Luciano committed:
Atualiza requirements
